pkg/api/http_server.go

```diff
@@ -288,7 +288,7 @@ func (hs *HTTPServer) applyRoutes() {
 	// then custom app proxy routes
 	hs.initAppPluginRoutes(hs.macaron)
 	// lastly not found route
-	hs.macaron.NotFound(hs.NotFoundHandler)
+	hs.macaron.NotFound(middleware.ReqSignedIn, hs.NotFoundHandler)
```

Comment:
Why is this change necessary?

Reply:
Some time in the past it was discovered that there was a security issue allowing users to bypass the login screen to a degree by pointing the browser to an unsupported route. 
https://github.com/grafana/grafana/pull/19118